lib/Service/PhishingDetection/ImapFlagCheck.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\Service\PhishingDetection;
+
+use Horde_Imap_Client;
+use OCA\Mail\PhishingDetectionResult;
+use OCP\IL10N;
+
+class ImapFlagCheck {
+	protected IL10N $l10n;
+
+	public function __construct(IL10N $l10n) {
+		$this->l10n = $l10n;
+	}
+
+	public function run(array $messageFlags) : PhishingDetectionResult {
+		$flaggedAsSpam = in_array(Horde_Imap_Client::FLAG_JUNK, $messageFlags) || in_array('junk', $messageFlags);
```

Comment:
nitpick: always add `true` as third argument of `in_array` to enforce strict type comparisons. We *shouldn't* receive anything but strings for the flags but better safe than sorry. Non-strict PHP comparisons are unpredictable.